modules/@apostrophecms/util/ui/src/util.js

```diff
@@ -184,24 +197,24 @@ export default () => {
   // DON'T try to find all the widgets. DO just enhance `el`.
   // This is a computer science principle known as "separation of concerns."
 
-  apos.util.runPlayers = function(el) {
-    const players = apos.util.widgetPlayers;
+  apos.util.runPlayers = function (el, newPlayer) {
```

Comment:
Should be `newPlayers`

Reply:
I still think we're fighting ghosts here, the real problem is players that replace the element completely, which we should document you should not do because it prevents tracking if players have been run (or your player must check that on its own if it insists on replacing the element)

But fine with this optimization if we name the parameter better

Reply:
I tested on players that are not replacing the element, but mounting (inserting child nodes). The player guard itself is mostly not working because the rapid event fire and DOM changes racing conditions. I spent a good amount of time debugging the whole. I think with this change, players are doing good on itself, eliminating the same tick executions during initialization. The next thing we might want to insect someday is the Vue context component `refresh` calls, that in some cases are spamming the refresh event - it should be probably better computed and/or slightly throttled.

Reply:
I came up with a better strategy I think. I like it more because it gives no "freedom" to pass any player outside of already registered, but still does improve the performance and guards against "player spamming".

Reply:
Did some more testing and debugging... and reverted the whole new players thing. The performance gain is questionable, we need a more clear strategy for improvement here.